UIPS/UIP-0131.md

```diff
@@ -0,0 +1,121 @@
+
+The protocol implements rate limiting, mandatory cooldowns, and minimum confirmation periods to prevent timing attacks. A monitoring client watches both chains for fraudulent tickets and automates violation reporting.
+
+TODO: Specify signature thresholds, timeouts, rate limits, and violation report format.
+
+### Arvo Integration
+
+The introduction of multi-chain state requires significant changes to Urbit's PKI verification system. Currently, %jael maintains Azimuth state by synthesizing data from Ethereum and the naive rollup. The addition of Base requires extending this system to handle three distinct sources of truth.
+
+The %base-watcher agent extends %jael's PKI verification capabilities, monitoring Base chain activity and updating Azimuth state accordingly. While the Ministry bridge protocol handles transfer sequencing and validation through its vault mechanism and oracle system, %jael remains responsible for maintaining a consistent view of the PKI state. This includes tracking point locations across chains and handling state rollbacks that may occur before bridge transfers are finalized.
+
+In cases of rollbacks, %jael must ensure that its state synthesis algorithm appropriately handles the reversion without compromising the integrity of cross-chain point locations. This becomes particularly important when points are in transit between chains.
+
+TODO: Detail exact state synthesis algorithm for handling multi-chain state reconciliation.
+
+### Base Implementation
+
+Base's optimistic rollup architecture introduces specific timing requirements for the return path of the bridge protocol. While points can begin operating immediately upon arrival on Base, returns to Ethereum require waiting for Base's 7-day challenge period to expire. This ensures the Embassy contract state is final before the Ministry vault releases points back to Ethereum.
+
+Oracle validation windows adapt to Base's block confirmation schedule, requiring a minimum of 15 L2 blocks before accepting transfer tickets. The Embassy contract implements a local timeout mechanism to handle potential state synchronization issues between Base and Ethereum during return transfers.
+
+TODO: Specify exact block confirmation thresholds and timeout parameters.
+
+Rationale
+---------
+
+The design establishes Ethereum as the senior chain to preserve the existing security model while enabling point mobility. Using galaxies as oracles leverages the network's existing power structure and economic incentives rather than introducing external validators. Points maintain complete functionality across chains through comprehensive state transfer.
```

Comment:
> Using galaxies as oracles leverages the network's existing power structure and economic incentives rather than introducing external validators.

Not obvious to me that there's broad consensus on whether this should be a role the galaxies should take on, whether they should be forced to take it on, or whether it should be a role exclusively galaxies can take on.

Reply:
As currently conceived this is a role that optionally galaxies may take on if they so chose. I would be of the mind that forcing anyone out of their current state (i.e. Ownership of their point on L1 Ethereum and no operational obligations) would be counter to the ethos of digital asset ownership. In the same way that the senate _could_ vote to disenfranchise any particular point's owner, but would not because that is counter to the idea of ownership, the senate should not abuse minority rights in forcing them out of the current state. Providing more enticing alternatives should absolutely be on the table though.